File: Runs/Java_repo_codegemma:7b/Test_7.java
public class Test_7 {

    public static int add(int a, int b) {
            return a + b;
        }

    public static int subtract(int a, int b) {
      return a - b;
    }

    public static int calculate(int x, int y, java.util.function.IntBinaryOperator op) {
      return op.applyAsInt(x, y);
    }

    public static int add(int a, int b) {
        return a + b;
      }
    
      public static int subtract(int a, int b) {
        return a - b;
      }
    
      public static int calculate(int a, int b, int (*func)(int, int)) {
        return func(a, b);
      }
    
      public static void main(String[] args) {
        int (*add_ptr)(int, int) = &add;
        System.out.println("Add: " + calculate(10, 5, add_ptr));
        System.out.println("Subtract: " + calculate(10, 5, &subtract));
      }

}

File: C_repo/test_7.c
#include <stdio.h>

// Tests how function pointers are translated.
// C function pointers map to Java Interfaces or Lambda expressions.

int add(int a, int b) {
    return a + b;
}

int subtract(int a, int b) {
    return a - b;
}

// 'op' is a function pointer
int calculate(int x, int y, int (*op)(int, int)) {
    return op(x, y);
}

int main() {
    int (*add_ptr)(int, int) = &add;
    
    printf("Add: %d\n", calculate(10, 5, add_ptr));
    printf("Subtract: %d\n", calculate(10, 5, &subtract));
    return 0;
}